### ISSUE TITLE
Error starting minikube on Windows (Error message is swallowed due to retry loop)

### ISSUE THREAD

**Issue by alexbrand** [kubernetes/minikube#405]
I've run into problems testing out Windows support on my dev machine. I am unable to diagnose what the actual issue is because the initial error is not logged due to the retry loop.

What seems to be happening is that the first attempt with a non-existent `.minikube` directory fails midway for some unknown reason, but it advances long enough so that the call to `api.Exists(constants.MachineName)` returns true on the next attempt.

Once this happens, `minikube` gets confused because it thinks the machine exists but fails later on trying to get the state of a non-existent machine:
```
PS C:\Users\abrand> minikube start  --logtostderr --show-libmachine-logs
Starting local Kubernetes cluster...
Creating CA: C:\Users\abrand\.minikube\certs\ca.pem
Creating client certificate: C:\Users\abrand\.minikube\certs\cert.pem
Running pre-create checks...
Creating machine...
(minikubeVM) Downloading C:\Users\abrand\.minikube\cache\boot2docker.iso from file://C:\Users\abrand\.minikube/cache/iso/minikube-0.5.iso...
I0726 18:47:54.940675    7516 cluster.go:69] Machine exists!
I0726 18:47:55.124184    7516 cluster.go:77] Machine state:  Error
I0726 18:47:55.125149    7516 cluster.go:69] Machine exists!
I0726 18:47:55.303738    7516 cluster.go:77] Machine state:  Error
E0726 18:47:55.304679    7516 start.go:78] Error starting host:  Error getting state for host: machine does not exist
```

I'd like to look into this more, but I am unable to build with `GOOS=windows`. Is there anything else I need to do to build `minikube` on Linux, targeting `windows`? I can spend some cycles on this.

**Comment by aaron-prindle** [comment]
I am getting the same errors attempting to run minikube on my Windows machine with Virtualbox.  @jimmidyson I believe you had mentioned that you had gotten minikube on Windows to work properly, did you happen to make any changes to minikube.  Also which driver were you using?  Thanks.

**Comment by alexbrand** [comment]
Found what I think was the problem, although I am not able to build and test this on windows.

**Comment by dlorenc** [comment]
What do you see when you try to build @alexbrand ? does `make cross` work for you?

**Comment by alexbrand** [comment]
Was able to build and test the fix in #406. I am happy to report that it runs on my Windows machine now 👍

**Comment by dlorenc** [comment]
Let's leave this open to make the retry loop not swallow errors. I can take that one.
